File: codeforces/1196/D2.cpp
#include <iostream>
#define REP(i, a, b) for (auto i = (a); i < (b); ++i)
using namespace std;

string gen(int n, int b) {
	string s(n, '0');
	REP(i, 0, n) {
		switch((i + b) % 3) {
			case 0:
				s[i] = 'R';
				break;
			case 1:
				s[i] = 'G';
				break;
			case 2:
			 s[i] = 'B';
			 break;
		}
	}
	return s;
}

signed main() {
	ios_base::sync_with_stdio(false);
	cin.tie(nullptr);

	int q; cin >> q;
	while (q--) {
		int n, k; cin >> n >> k;
		string s; cin >> s;
		
		int ans = n;
		REP(b, 0, 3) {
			string t = gen(n, b);
			int diff = 0;
			REP(i, 0, k)
				if (s[i] != t[i])
					 ++diff;
			ans = min(ans, diff);
			REP(i, k, n) {
				if (s[i - k] != t[i - k]) --diff;
				if (s[i] != t[i]) ++diff;
				ans = min(ans, diff);
			}
		}
		cout << ans << '\n';
	}
}

File: codeforces/1196/D2.py
import sys
input = sys.stdin.readline

q = int(input())
for _ in range(q):
	n, k = map(int, input().split())
	s = input()

	a = [0] * n
	for i in range(n):
		if s[i] == "R":
			a[i] = 0
		if s[i] == "G":
			a[i] = 1
		if s[i] == "B":
			a[i] = 2

	ans = n
	for b in range(3):
		diff = 0

		for i in range(k):
			if a[i] != (i + b) % 3:
				diff += 1
		ans = min(ans, diff)

		for j in range(k, n):
			if a[j - k] != (j - k + b) % 3:
				diff -= 1
			if a[j] != (j + b) % 3:
				diff += 1
			ans = min(ans, diff)

	print(ans)